pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -162,8 +156,8 @@ func ApplySearchHeadCluster(ctx context.Context, client splcommon.ControllerClie
 	deployer := &enterpriseApi.Deployer{}
 	err = client.Get(context.TODO(), namespacedName, deployer)
 	if err == nil {
-		// when user creates both cluster manager and index cluster yaml file at the same time
-		// cluser manager status is not yet set so it will be blank
+		// when user creates both SHC and deployer yaml file at the same time
```

Comment:
@line 154, what happens, if the deployer reference is not configured intentionally?

Reply:
We invalidate the SHC spec if deployerRef is not configured.